### PR TITLE
Feature/runner

### LINKED ISSUE #27: Add ability to run user's code.
Malware security is to be implemented as well.

### LINKED ISSUE #36: Add restrictions to runned code.
Add some white list of linux syscalls.

### PR DESCRIPTION
**Pull request by OleksandrStrashnenko**
There are still opened issues referring runner module. But I would like to introduce some raw version.

### CODE REVIEW

File: runner/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+slog = "2.7.0"
+interprocess = "1.1.1"
+serde = {version = "1.0.130", features=["derive"], optional = true }
+serde_json = {version = "^1.0.9", optional = true}
+
+[dependencies.compiler]
+path = "../compiler"
+
+[build-dependencies]
+bindgen = "0.59.1"
```
Comment (AlexanderSatana) [inline runner/Cargo.toml:40]:
```suggestion
bindgen = "0.59.1"
```suggestion
bindgen = "0.59.1"
```suggestion
bindgen = "0.59.1"
```
```
```

Comment (AlexanderSatana) [inline runner/Cargo.toml:44]:
```suggestion
bindgen = "0.59.1"

```

**Review by virtustron — APPROVED** [review]

**Review by AndriiSerbeniuk — APPROVED** [review]

File: runner/build.rs [hunk 2]
```diff
@@ -0,0 +1,16 @@
+//extern crate bindgen;
+
+fn main() {
```
Comment (valentynfaychuk) [inline runner/build.rs:3]: is it possible to get rid of build.rs script (if it is of no use)?

File: runner/lib/filter/mod.rs [hunk 3]
```diff
@@ -0,0 +1,27 @@
+use std::fs::{File};
+use std::path::Path;
+use std::env;
+use seccompiler::{compile_from_json, BpfProgram, TargetArch};
+
+const FILTERS_CONFIG_PATH: &str = "config/filters.json";
```
Comment (ValentynFk) [inline runner/lib/filter/mod.rs:6]: is it possible to make the path configurable, can we use the environment variable?

File: runner/lib/filter/mod.rs [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+            String::from(FILTERS_CONFIG_PATH)
+        }
+    };
+    assert!(Path::new(config_path.as_str()).exists());
+    let mut filters = compile_from_json(
+        File::open(config_path.as_str()).unwrap(), 
+        TargetArch::x86_64
+    ).expect("Cannot compile filters from json");
+    // TODO: use preset string as func param
+    let bpf_prg: BpfProgram = filters.remove("main_thread").unwrap();
```
Comment (valentynfaychuk) [inline runner/lib/filter/mod.rs:24]: seems like there are lot's of unwraps, I assume all calls that may fail should use `?` approach
isn't this a reason why you use `Result<..., ...>` as a return type?

**Review by valentynfaychuk — APPROVED** [review]

**Pull request merged**